src/rapidgzip/gzip/GzipReader.hpp

```diff
@@ -41,7 +44,27 @@ class GzipReader :
 public:
     explicit
     GzipReader( UniqueFileReader fileReader ) :
-        m_bitReader( std::move( fileReader ) )
+        m_fileReader( ensureSharedFileReader( std::move(fileReader) ) ),
+        m_bitReader( m_fileReader->clone() )
+    {}
+
+    explicit
+    GzipReader( const GzipReader& other ) :
+        m_fileReader( other.m_fileReader->clone() ),
```

Comment:
If `m_fileReader` was wrapped in some copy-able class that simply calls clone, then this whole copy-constructor could be defaulted (assuming the shared pointers should not be deep-copied, see my other comments)!